MekHQ/src/mekhq/campaign/autoresolve/acar/manager/PhasePreparationManager.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2024 - The MegaMek Team. All Rights Reserved.
+ *
+ *  This file is part of MekHQ.
+ *
+ *  MekHQ is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  MekHQ is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mekhq.campaign.autoresolve.acar.manager;
+
+import mekhq.campaign.autoresolve.acar.SimulationManager;
+
+public record PhasePreparationManager(SimulationManager simulationManager) implements SimulationManagerHelper {
+
+    public void managePhase() {
+        clearActions();
+        switch (game().getPhase()) {
```

Comment:
This would be a lot more succinct as an enhanced switch:

```
switch (game().getPhase()) {
            case DEPLOYMENT, SBF_DETECTION, MOVEMENT, FIRING -> {
                resetEntityPhase();
                simulationManager.initiativeHelper.determineTurnOrder(game().getPhase());
            }
            case INITIATIVE -> clearActions();
            case END, VICTORY, default -> {
                clearReports();
                clearActions();
            }
        }
```

Reply:
but would also be wrong, as initiative also run `resetEntityPhase` and the determineTurnOrder.

Reply:
Ahhhh, I hate reviewing through GitHub, you're quite right

Reply:
you can use split or unified view, both have some ups and downs for revieweing and make it a bit easier depending on the type of difficulty you are having. I particularly like using github for reviewing code.